packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -192,25 +318,73 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
             "Commitment has expired"
         );
 
-        CreateLoanArgs memory createLoanArgs;
-        createLoanArgs.marketId = _marketId;
-        createLoanArgs.lendingToken = _tokenAddress;
-        createLoanArgs.principal = _principal;
-        createLoanArgs.duration = _loanDuration;
-        createLoanArgs.interestRate = _interestRate;
+        require(
+            commitment.maxPrincipalPerCollateralAmount == 0 ||
+                _collateralAmount *
+                    (commitment.maxPrincipalPerCollateralAmount) >=
+                _principalAmount * PRINCIPAL_PER_COLLATERAL_EXPANSION_FACTOR,
+            "Insufficient collateral"
+        );
```

Comment:
Wouldn't this require fail if the collateral is USDC & ETH is Loan Amount. For Ex: 2000 USDC as collateral for 1 ETH loan Volume ?

Reply:
See latest unit tests for getRequiredCollateral_test  .. now using 16 decimals